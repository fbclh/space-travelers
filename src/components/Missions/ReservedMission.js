import PropTypes from 'prop-types';
import { useDispatch } from 'react-redux';
import { leaveMission } from '../../redux/missions/missions';

export default function ReservedMission({ missionName, id, wikipedia }) {
  const dispatch = useDispatch();

  const handleLeaveMission = () => {
    dispatch(leaveMission(id));
  };

  const handleWikipedia = () => {
    window.open(wikipedia, '_blank');
  };

  return (
    <tr>
      <td className="td">{missionName}</td>
      <td className="td td-button">
        <button
          className="btn btn-outline-danger"
          type="button"
          onClick={handleLeaveMission}
        >
          Leave Mission
        </button>
      </td>
      <td className="td td-button">
        <button
          className="btn btn-secondary"
          type="button"
          onClick={handleWikipedia}
        >
          Read More
        </button>
      </td>
    </tr>
  );
}

ReservedMission.propTypes = {
  missionName: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired,
  wikipedia: PropTypes.string.isRequired,
};
