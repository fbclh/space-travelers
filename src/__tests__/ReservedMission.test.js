import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import ReservedMission from '../components/Missions/ReservedMission';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <ReservedMission />
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
