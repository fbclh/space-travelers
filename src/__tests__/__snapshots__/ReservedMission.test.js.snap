// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<tr>
  <td
    className="td"
  />
  <td
    className="td td-button"
  >
    <button
      className="btn btn-outline-danger"
      onClick={[Function]}
      type="button"
    >
      Leave Mission
    </button>
  </td>
  <td
    className="td td-button"
  >
    <button
      className="btn btn-secondary"
      onClick={[Function]}
      type="button"
    >
      Read More
    </button>
  </td>
</tr>
`;
